src/main/java/com/fasterxml/jackson/databind/ser/PropertyBuilder.java

```diff
@@ -204,11 +204,6 @@ protected BeanPropertyWriter buildWriter(SerializerProvider prov,
             break;
         case CUSTOM: // new with 2.9
             valueToSuppress = prov.includeFilterInstance(propDef, inclV.getValueFilter());
-            if (valueToSuppress == null) { // is this legal?
-                suppressNulls = true;
-            } else {
-                suppressNulls = prov.includeFilterSuppressNulls(valueToSuppress);
```

Comment:
We will likely want to then deprecate `includeFilterSuppressNulls` in `SerializerProvider` (to be removed from 3.0, kept in 2.x)